Functions/xASL_bids_PrepareDRO.m

```diff
@@ -0,0 +1,96 @@
+%               droTestPatientSource = fullfile(TestRepository,'UnitTesting','dro_files','test_patient');
+%               droTestPatient = fullfile(TestRepository,'UnitTesting','working_directory','test_patient');
+%               xASL_Copy(droTestPatientSource,fullfile(droTestPatient,'rawdata',droSubject));
+%               xASL_bids_PrepareDRO(droTestPatient,droSubject);
+%               
+% __________________________________
+% Copyright 2015-2021 ExploreASL
+
+    % Check input arguments
+    if nargin<2
+        droSubject = 'sub-Sub1'; % DEFAULT
+    end
+
+    % Directory definitions
+    perfDirectory = fullfile(droTestPatient,'rawdata',droSubject,'perf');
+    groundTruthDirectory = fullfile(droTestPatient,'rawdata',droSubject,'ground_truth');
+    anatDirectory = fullfile(droTestPatient,'rawdata',droSubject,'anat');
+
+    % Rename asl to perf
+    xASL_Move(fullfile(droTestPatient,'rawdata',droSubject,'asl'),perfDirectory);
+
+    % Rename ASL
+    xASL_Move(fullfile(perfDirectory,'001_asl.json'),fullfile(perfDirectory,[droSubject,'_asl.json']));
+    xASL_Move(fullfile(perfDirectory,'001_asl.nii.gz'),fullfile(perfDirectory,[droSubject,'_asl.nii.gz']));
+    xASL_Move(fullfile(perfDirectory,'001_aslcontext.tsv'),fullfile(perfDirectory,[droSubject,'_aslcontext.tsv']));
+
+    % Add M0 from ground_truh
+    xASL_Move(fullfile(groundTruthDirectory,'002_ground_truth_m0.json'),fullfile(perfDirectory,[droSubject,'_m0scan.json']));
+    xASL_Move(fullfile(groundTruthDirectory,'002_ground_truth_m0.nii.gz'),fullfile(perfDirectory,[droSubject,'_m0scan.nii.gz']));
+
+    % Rename T1w
+    xASL_Move(fullfile(anatDirectory,'003_anat.json'),fullfile(anatDirectory,[droSubject,'_T1w.json']));
+    xASL_Move(fullfile(anatDirectory,'003_anat.nii.gz'),fullfile(anatDirectory,[droSubject,'_T1w.nii.gz']));
+
+    % Remove the ground truth files
+    xASL_delete(groundTruthDirectory,true);
+
+    % Read ASL JSON file and add the M0Type field
+    jsonASL = spm_jsonread(fullfile(perfDirectory,[droSubject,'_asl.json']));
+    jsonASL.M0Type = "Separate";
+    spm_jsonwrite(fullfile(perfDirectory,[droSubject,'_asl.json']),jsonASL);
+
+    % Read M0 JSON file and add the IntendedFor field
+    jsonM0 = spm_jsonread(fullfile(perfDirectory,[droSubject,'_m0scan.json']));
+    jsonM0.IntendedFor = "perf/sub-Sub1_asl.nii.gz";
+    spm_jsonwrite(fullfile(perfDirectory,[droSubject,'_m0scan.json']),jsonM0);
+
+    nameDRO = "DRO_Digital_Reference_Object";
+    [jsonDescription] = createDatasetDescriptionTemplate(nameDRO);
+
+    % Write file
+    spm_jsonwrite(fullfile(droTestPatient,'rawdata','dataset_description.json'),jsonDescription);
+
+
+end
+
+%% Create dataset_description.json template
+function [json] = createDatasetDescriptionTemplate(name)
+
+    % Create dummy dataset_description.json
+    json = struct;
+    if nargin > 0
+        json.Name = name;
+    else
+        json.Name = "RandomText";
+    end
+    json.BIDSVersion = "1.5.0";
```

Comment:
Call bidsPar = xASL_bids_Config() and use bidsPar.BIDSVersion

Reply:
I added this in 3e2b8e1. Is there a script that creates a dataset description template already, like I do in the subfunction `createDatasetDescriptionTemplate`? If not, maybe such a helper function could be useful for other applications as well.

Reply:
Thanks. Regarding Default description - yes, there should be a function. And this function better has some normal text and not RandomText :) My fault mainly as I used these for the ImportBIDS to have some values for testing. But according to BIDS, only name and version is required.

So lets leave this as it is now and I have created a new issue to fix it for all cases. #443